File: BKJN_CT/SmallLighter/1697.js
const filePath =
  process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt';
const input = require('fs')
  .readFileSync(filePath)
  .toString()
  .trim()
  .split('\n');

const [n, k] = input.shift().split(' ').map(Number);

let visited = new Array(100001).fill(false);

const bfs = (v) => {
  let q = [];
  q.push(v);

  visited[v] = true;
  let second = 0;

  while (q.length) {
    const cycle = q.length;
    for (let i = 0; i < cycle; i++) {
      const x = q.shift();
      if (x === k) return second;
      for (let nx of [x - 1, x + 1, 2 * x]) {
        if (nx >= 0 && nx <= 100000 && !visited[nx]) {
          visited[nx] = true;
          q.push(nx);
        }
      }
    }
    second++;
  }
};

console.log(bfs(n));

File: BKJN_CT/SmallLighter/1697.py
import sys
from collections import deque

sys.stdin = open("./BKJN_CT/SmallLighter/input.txt", "r")

input = sys.stdin.readline

N, K = map(int, input().split())

# 가장 빠른 시간 = BFS

visited = [False] * 100001

def BFS(v):
    q = deque([v])
    ans = 0

    while(q):
        length = len(q)
        for _ in range(length):
            x = q.popleft()
            if x == K:
                print(ans)
                break
            for nx in (x -1 , x + 1, x * 2):
                if 0 <= nx < 100001 and not visited[nx]:
                    visited[nx] = True
                    q.append(nx)
        ans += 1

BFS(N)